sycl/doc/design/CompileTimeProperties.md

```diff
@@ -596,6 +595,66 @@ we do not allow non-fundamental types, how do we represent properties like
 allow `std::tuple`, where the type of each element is one of the fundamental
 types listed above.
 
+### IR representation via `@llvm.global.annotations`
+
+Properties that are implemented using
+`[[__sycl_detail__::add_ir_attributes_global_variable()]]`, are represented in 
+LLVM IR as elements in the global variable `@llvm.global.annotations`.
+`@llvm.global.annotations` is an array of `{ ptr, ptr, ptr, i32, ptr}` structs,
+where each struct represents an annotation on a global pointer. The values of
+each field of the struct are as follows:
+
+* pointer to global object to be annotated, global variable in our case. If 
+there are nested annotations on the global variable, constant `getelementptr`
+instructions can be used to specify at which level the annotation is applied.
+* pointer to global string, representing the annotation. Each global string is
+a list of `{name:value}` pairs from the
+`[[__sycl_detail__::add_ir_attributes_global_variable()]]` annotation. Property
+values are converted to strings in the same way as described above, except that
+the `nullptr` value is represented as an empty string, `{name}`.
+* pointer to global string, representing the path of the source file where the
+annotation was applied.
+* i32 integer representing the line number in the source file where the
+annotation was applied. 
+* ptr to annotation arguments, `null` pointer in our case 
+
+```cpp
+device_global<fpga_mem<int[4], decltype(properties(word_size<2>))>, decltype(properties(host_access_none))> dg;
+```
+
+Below is what the LLVM IR will look like for the SYCL code above.
```

Comment:
The code doesn't actually show how the annotation is applied. I would suggest providing pseudocode for that in this document as well.

Reply:
I added a pseudo header definition of `fpga_mem` similar to one for `device_global`. Now there is full transparency on how that line gets translated into the IR below.